### ISSUE TITLE
[Code] SubdivisionModifier UV Support

### ISSUE THREAD

**Issue by centerionware** [mrdoob/three.js#7903]
Uhm, Well, I've begun adding(trying to) UV support that's working for me with the Subdivision Modifier. It only works with 1 level of subdivision in my test.
The modified source is --redacted. The first attempt was incredibly naive. The link in the following post works.
(If anyone wants to help test, validate, then submit it as a patch, if it ever proves the test of time..?)

My test unit did use a DDS texture.. stupid backwards upside down uv mapping. I don't know if that affected the final results.

Hope this is helpful.

**Comment by centerionware** [comment]
so, I went and cleaned up the idea a bit more, and cleaned up the math.
http://www.centerionware.com/javascript/three/examples/js/modifiers/testing_SubdivisionModifier.js
It works @ 1, 2, and 3 subdivisions, works with PNG images, and DDS (Although I had to make the DDS upside down or things went badly, as tends to happen with dds images.)
 This version is about 20 lines extra on top of the original subdivisionmodifier, most being variable declarations or variable reference declarations, 2 tiny new functions (midpoint(a, b), and newuv(array, a, b, c) (keeping in style with newface)). It creates 3 new uv's for the 3 new vertices being created on each subdiv, then calls newUv 4 times with the layout dicated by the order the faces are drawn and how they are drawn..
It also contains a big picture in comments that helped me visualize how the triangles were being subdivided so it was easier to figure out how things fit together nicely.
